src/main/java/org/reaktivity/specification/mqtt/internal/MqttFunctions.java

```diff
@@ -201,37 +225,28 @@ public MqttDataExBuilder correlationInfo(
             return this;
         }
 
-        public byte[] build()
+        public MqttDataExBuilder userProperty(
+            String name,
+            String value)
         {
-            final MqttDataExFW dataEx = dataExRW.build();
-            final byte[] array = new byte[dataEx.sizeof()];
-            dataEx.buffer().getBytes(dataEx.offset(), array);
-            return array;
-        }
-    }
-
-    public static final class MqttEndExBuilder
-    {
-        private final MqttEndExFW.Builder endExRW;
-
-        private MqttEndExBuilder()
-        {
-            MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[1024 * 8]);
-            this.endExRW = new MqttEndExFW.Builder().wrap(writeBuffer, 0, writeBuffer.capacity());
-        }
-
-        public MqttEndExBuilder typeId(
-            int typeId)
-        {
-            endExRW.typeId(typeId);
+            if (value == null)
+            {
+                dataExRW.propertiesItem(p -> p.key(name)
+                                              .value((String) null));
+            }
+            else
+            {
+                dataExRW.propertiesItem(p -> p.key(name)
+                                              .value(value));
+            }
```

Comment:
Doesn't the `else` clause work for both cases, given that `value` parameter is of type `String`?